src/frozen_solvers.jl

```diff
@@ -0,0 +1,260 @@
+
+"""
+    ManifoldEuler
+
+The manifold Euler algorithm for problems in the [`ExplicitManifoldODEProblemType`](@ref)
+formulation.
+"""
+struct ManifoldEuler{TM<:AbstractManifold,TR<:AbstractRetractionMethod} <:
+       OrdinaryDiffEqAlgorithm
+    manifold::TM
+    retraction_method::TR
+end
+
+alg_order(::ManifoldEuler) = 1
+
+"""
+    ManifoldEulerCache
+
+Cache for [`ManifoldEuler`](@ref).
+"""
+struct ManifoldEulerCache <: OrdinaryDiffEqMutableCache end
+
+"""
+    ManifoldEulerConstantCache
+
+Cache for [`ManifoldEuler`](@ref).
+"""
+struct ManifoldEulerConstantCache <: OrdinaryDiffEqConstantCache end
+
+function alg_cache(
```

Comment:
This to me feels quite long for then calling a parameterless function. Can this maybe be shortened? (maybe it can not)

Reply:
I don't think this can be shortened, I'd leave this for now.

Reply:
Ok.